addons/mail/static/src/owl/store/messaging_actions.js

```diff
@@ -3326,7 +3327,7 @@ const actions = {
         const threadAttachmentLocalIds = thread.attachmentLocalIds;
         if (!threadAttachmentLocalIds || !threadAttachmentLocalIds.includes(attachmentLocalId)) {
             dispatch('_updateThread', threadLocalId, {
-                attachmentLocalIds: threadAttachmentLocalIds.concat([attachmentLocalId]),
+                attachmentLocalIds: [attachmentLocalId].concat(threadAttachmentLocalIds),
```

Comment:
ok, I may break ordering of attachments in my entity branch, so will need to be keep an eye on that ;)